File: backend/src/main/java/com/example/entity/KeHoach.java
package com.example.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.NaturalId;

import java.util.Date;

@Entity
@Getter
@Setter
public class KeHoach {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long maKeHoach;
    private String tenKeHoach;
    private Date hanHoanThanh;
    private Integer soLuong;
    private String diaDiem;
    private String noiDung;
    @OneToOne
    @JoinColumn(name = "ma_thong_tin_phan_anh", referencedColumnName = "maThongTinPhanAnh")
    private ThongTinNguoiDanPhanAnh thongTinNguoiDanPhanAnh;
}

File: fronted/src/models/KeHoach.ts
import BaseResponse from "./BaseResponse";

export interface KeHoachResponse extends BaseResponse {
    maKeHoach: number
    tenKeHoach: string
    hanHoanThanh: string
    soLuong: number
    diaDiem: string
    noiDung: string
    maThongTinPhanAnh: number
}

export interface KeHoachRequest {
    maKeHoach?: number
    tenKeHoach: string
    hanHoanThanh: string
    soLuong: number
    diaDiem: string
    noiDung: string
    maThongTinPhanAnh?: number
}